x/axelarcork/types/proposal.go

```diff
@@ -0,0 +1,167 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	ProposalTypeAddManagedCellarIDs    = "AddManagedCellarIDs"
+	ProposalTypeRemoveManagedCellarIDs = "RemoveManagedCellarIDs"
+	ProposalTypeScheduledCork          = "ScheduledCork"
+	ProposalTypeCommunitySpend         = "CommunitySpend"
+)
+
+var _ govtypes.Content = &AddManagedCellarIDsProposal{}
+var _ govtypes.Content = &RemoveManagedCellarIDsProposal{}
+var _ govtypes.Content = &ScheduledCorkProposal{}
+var _ govtypes.Content = &CommunityPoolSpendProposal{}
+
+func init() {
+	govtypes.RegisterProposalType(ProposalTypeAddManagedCellarIDs)
+	govtypes.RegisterProposalTypeCodec(&AddManagedCellarIDsProposal{}, "sommelier/AddManagedCellarIDsProposal")
+
+	govtypes.RegisterProposalType(ProposalTypeRemoveManagedCellarIDs)
+	govtypes.RegisterProposalTypeCodec(&RemoveManagedCellarIDsProposal{}, "sommelier/RemoveManagedCellarIDsProposal")
+
+	govtypes.RegisterProposalType(ProposalTypeScheduledCork)
+	govtypes.RegisterProposalTypeCodec(&ScheduledCorkProposal{}, "sommelier/ScheduledCorkProposal")
+}
+
+func NewAddManagedCellarIDsProposal(title string, description string, cellarIds *CellarIDSet) *AddManagedCellarIDsProposal {
+	return &AddManagedCellarIDsProposal{
+		Title:       title,
+		Description: description,
+		CellarIds:   cellarIds,
+	}
+}
+
+func (m *AddManagedCellarIDsProposal) ProposalRoute() string {
+	return RouterKey
+}
+
+func (m *AddManagedCellarIDsProposal) ProposalType() string {
+	return ProposalTypeAddManagedCellarIDs
+}
+
+func (m *AddManagedCellarIDsProposal) ValidateBasic() error {
+	if err := govtypes.ValidateAbstract(m); err != nil {
+		return err
+	}
+
+	if len(m.CellarIds.Ids) == 0 {
+		return fmt.Errorf("can't have an add prosoposal with no cellars")
+	}
+
+	return nil
+}
```

Comment:
Because we don't have a reference to the Keeper in ValidateBasic checks, we can only do checks that don't require chain state. That's why we use the ChainConfigurationByNameAndID in every handler.

Reply:
Should it be a check inside of `ChainConfigurationByNameAndID` then? My concern is someone could pass in `1` and `avalanche` (unrealistic example but you get my point) in a proposal or strategy call and we don't really know what their actual intent was. If anything the name is more realistic to default to, but that method checks ID first.

Reply:
I think I am confused? If you provide both options to ChainConfigurationByNameAndID, then it errors if they don't match. What you are describing is what it does